### DISCUSSION TITLE
uart.c 存放位置

### DISCUSSION THREAD

**Discussion by ghost** [bol-edu/HLS-SOC-Discussions#154] [Q&A]
想問為什麼uart.c的firmware code是放在.mprj(mprj : ORIGIN = 0x30000000)，MMIO 0x30000000位置不是放uart硬體嗎?為什麼可以放這段firmware code。
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/119398292/a9c67f98-abc4-4852-8a41-a620e6ab71e9)

**Comment by LingChiYang** [comment]
同學好

存在這個位置確實不太必要。但對於Lab6我想沒有影響，uart的code並不會送到user project area。
